amy/autoemails/actions.py

```diff
@@ -1133,3 +1134,66 @@ def get_additional_context(self, objects, *args, **kwargs):
             )
 
         return context
+
+
+class NewConsentRequiredAction(BaseAction):
+    """
+    Action for asking users to consent to newly required terms. This email
+    should be sent when a new required Term is created.
+
+    How to use it:
+
+    >>> triggers = Trigger.objects.filter(active=True,
+                                          action='consent-required-new')
+    >>> for trigger in triggers:
+    ...     action = NewConsentRequiredAction(
+    ...         trigger=trigger,
+    ...         objects=dict(terms=terms),
+    ...     )
+    ...     launch_at = action.get_launch_at()
+    ...     job = scheduler.enqueue_in(launch_at, action)
+    """
+
+    launch_at = timedelta(hours=1)
+
+    def get_launch_at(self):
+        return self.launch_at
+
+    def recipients(self) -> Optional[str]:
+        """Assuming self.context is ready, overwrite email's recipients
+        with selected ones."""
+        try:
+            return self.context["all_emails"]
+        except (AttributeError, KeyError):
+            return None
+
+    def all_recipients(self) -> str:
+        """If available, return string of all recipients."""
+        # TODO what is this versus recipients?
+        try:
+            terms = self.context_objects["terms"]
+            # TODO: Dependent on unset consents being stored
+            person_emails = list(
+                Consent.objects.filter(
```

Comment:
So this sends one email to multiple recipients. Can you check if that would result in one email or in one email per recipient? We'd rather not reveal people's email addresses.

Reply:
I messaged @pbanaszkiewicz  individually, but I'm going to summarize what we spoke about here:
- This appears to send 1 emaill with a very long to message to all the users included
- The way to avoid that is: https://anymail.readthedocs.io/en/stable/sending/templates/#batch-send
-  This appears to still send one email until it reaches mailgun. Mailgun handles separating them into multiple emails, so there's no way to double check that everything is ok without a mailgun sandbox. 
- I also hit a column length issue because my recipients list is so long. 

For now I'm going to:
- preserve the work I did to try to do bulk emails with Mailgun in another branch
- Convert this branch to manually creating an individual email per user so we can be sure it works correctly

What would be helpful in the future:
- Getting a sandbox mailgun account to test out the functionality and/or putting the mailgun account in AMY to sandbox mode so that we can test this once.

Reply:
@pbanaszkiewicz saving the pr here: https://github.com/lauryndbrown/amy/compare/1828/email-user-when-new-required-consent-to-approve...lauryndbrown:lb/trying-to-mass-email-correctly?expand=1